src/main/scala/org/constellation/checkpoint/CheckpointAcceptanceService.scala

```diff
@@ -333,29 +344,44 @@ class CheckpointAcceptanceService[F[_]: Concurrent: Timer](
   def acceptObservations(cb: CheckpointBlock, cpc: Option[CheckpointCache] = None): F[Unit] =
     cb.observations.toList.traverse(o => observationService.accept(o, cpc)).void
 
-  def acceptTransactions(cb: CheckpointBlock, cpc: Option[CheckpointCache] = None): F[Unit] = {
+  def acceptTransactions(
+    cb: CheckpointBlock,
+    cpc: Option[CheckpointCache] = None,
+    txsHashToFilter: Seq[String] = Seq.empty
+  ): F[Unit] = {
     def toCacheData(tx: Transaction) = TransactionCacheData(
       tx,
       Map(cb.baseHash -> true),
       cbBaseHash = Some(cb.baseHash)
     )
 
-    val insertTX =
-      LiftIO[F].liftIO {
-        cb.transactions.toList
-          .map(tx ⇒ (tx, toCacheData(tx)))
-          .traverse {
-            case (tx, txMetadata) =>
-              dao.transactionService.accept(txMetadata, cpc) >> transferIfNotDummy(tx)
-          }
-          .void
+    cb.transactions.toList
+      .filterNot(h => txsHashToFilter.contains(h.hash))
+      .map(tx => (tx, toCacheData(tx)))
+      .traverse {
+        case (tx, txMetadata) => LiftIO[F].liftIO(dao.transactionService.accept(txMetadata, cpc)) >> transfer(tx)
```

Comment:
you can use `transactionService` directly, especially if you have this dependency ready to use